### PR TITLE
Utility: release build archiving (zip)

### PR DESCRIPTION
**Pull request by jonpas**
- Utility `hemtt zip`
  - Usable build step as `@zip`
- Archive name configurability (`@zip <name>`?) #17 
  - Arguments support for utilities (and templating)

### CODE REVIEW

**Comment by jonpas** [comment]
Will change to utility. Related to #15.

**Comment by jonpas** [comment]
Going to wait for build refactor and templating to get merged.

**Comment by DerZade** [comment]
Would love to have this feature. :)
As far as I can tell both _build refactor_ and _templating_ are merged, so is there any other reason why this isn't?

**Comment by jonpas** [comment]
@DerZade read original comment.

**Comment by jonpas** [comment]
I used their example [here](https://github.com/mvdnes/zip-rs/blob/master/examples/write_dir.rs) for implementation. Might have missed something?

**Comment by synixebrett** [comment]
I don't see anything you missed, it works fine on Windows 10 using the built in Zip explorer. I think this just might be the zip library.

**Comment by jonpas** [comment]
@synixebrett seems to work fine on both of my Arch machines. `UnZip 6.00 of 20 April 2009, by Info-ZIP.`

What unzip does your Ubuntu use?

I will try that directory as it would simplify things.

Also random though, unit tests on Travis might be a good idea!

**Comment by jonpas** [comment]
Found the difference, looks like my unzip was juts handling it correctly and automatically creating folders.

Without `add_directory`:
```
$ unzip testName.zip
Archive:  testName.zip
zip-rs
  inflating: @TST/keys/TST.bikey
  inflating: @TST/mod.cpp
  inflating: @TST/addons/TST_example.pbo.0.1.0.0.bisign
  inflating: @TST/addons/TST_example.pbo
  inflating: @TST/addons/TST_main.pbo
  inflating: @TST/addons/TST_common.pbo
  inflating: @TST/addons/TST_main.pbo.0.1.0.0.bisign
  inflating: @TST/addons/TST_common.pbo.0.1.0.0.bisign
```
With `add_directory`:
```
$ unzip testName.zip
Archive:  testName.zip
zip-rs
warning:  stripped absolute path spec from /
mapname:  conversion of  failed
   creating: @TST/
   creating: @TST/keys/
  inflating: @TST/keys/TST.bikey
  inflating: @TST/mod.cpp
   creating: @TST/addons/
  inflating: @TST/addons/TST_example.pbo.0.1.0.0.bisign
  inflating: @TST/addons/TST_example.pbo
  inflating: @TST/addons/TST_main.pbo
  inflating: @TST/addons/TST_common.pbo
  inflating: @TST/addons/TST_main.pbo.0.1.0.0.bisign
  inflating: @TST/addons/TST_common.pbo.0.1.0.0.bisign
```

So this should fix it for your unzip as well @synixebrett, if you could try.

There is still a warning and mapname conversion failed error, no clue why yet.

File: docs/utilities.md [hunk 1]
```diff
@@ -23,3 +23,11 @@ Spanish          2015  75%
 Hungarian        1558  58%
 Turkish             3   0%
 ```
+
+# zip
+
+The zip utility will zip the current release into a .zip file. The zip filename will be `{{name}}_{{version}}` unless a name is provided.
+
+`hemtt zip` => `ace3_1.2.3.zip`
```
Comment (jonpas) [inline docs/utilities.md]: It would be `ace_1.2.3.zip`, because prefix of ACE3 ia just ace. :D

**Comment by synixebrett** [comment]
> Also random though, unit tests on Travis might be a good idea!
I've thought of unit tests, I would just like to get HEMTT a bit more flushed out and settled, so we aren't constantly adjusting tests as new features are brought in.

**Comment by synixebrett** [comment]
This does work better with add_directory; it did not fix it with UnZip <= 3.0 (2008). Which unfortunately is still in use.

**Pull request merged**